File: apis/java/src/main/java/com/github/zachdeibert/codetanks/api/model/MoveData.java
package com.github.zachdeibert.codetanks.api.model;

public final class MoveData extends LoginTokenContainer {
	public enum ForwardDirection {
		forward, backward
	}

	public enum StrafeDirection {
		left, right
	}

	public enum TurnDirection {
		clockwise, counterclockwise
	}

	public static final class MoveInfo<T> {
		public T direction;
		public double rate;
		
		public MoveInfo(T direction, double rate) {
			this.direction = direction;
			this.rate = rate;
		}
	}

	public MoveInfo<ForwardDirection> forward;
	public MoveInfo<StrafeDirection> strafe;
	public MoveInfo<TurnDirection> turn;
}

File: model/MoveData.cs
using System;
using Newtonsoft.Json;

namespace Com.GitHub.ZachDeibert.CodeTanks.Model {
    public sealed class MoveData : LoginTokenContainer {
        public enum ForwardDirection {
            [JsonProperty("forward")]
            Forwards,
            [JsonProperty("backward")]
            Backwards
        }

        public enum StrafeDirection {
            [JsonProperty("left")]
            Left,
            [JsonProperty("right")]
            Right
        }

        public enum TurnDirection {
            [JsonProperty("clockwise")]
            ClockWise,
            [JsonProperty("counterclockwise")]
            CounterClockWise
        }

        public sealed class MoveInfo<T> {
            [JsonProperty("direction")]
            public T Direction;
            [JsonProperty("rate")]
            public double Rate;
        }

        [JsonProperty("forward")]
        public MoveInfo<ForwardDirection> Forward;
        [JsonProperty("strafe")]
        public MoveInfo<StrafeDirection> Strafe;
        [JsonProperty("turn")]
        public MoveInfo<TurnDirection> Turn;
    }
}
